File: 746_Min Cost Climbing Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] stair = new int[2];
        
        // init: stair[0] = stair[1] = 0
        for (int i = 2; i <= cost.length; i++) {
            int tmp = Math.min(cost[i-2]+stair[0], cost[i-1]+stair[1]);
            stair[0] = stair[1];
            stair[1] = tmp;
        }
        return stair[1];
    }
}

File: 746_Min Cost Climbing Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        curs = nexts = 0
        for c in cost:
            # cost for the next 2 steps from here = curs + c
            # cost for the next step = min(nexts, curs + c)
            # cost for the next next step = curs + c
            curs, nexts = min(nexts, curs + c), curs + c
        return curs